### ISSUE TITLE
Cannot use combineK to combine Endpoints

### ISSUE THREAD

**Issue by custommonkey** [finagle/finch#1564]
Trying to use combineK from SemigroupK results in an error `value <+> is not a member of io.finch.Endpoint[F,String]`

### Example failure

```scala
  import cats.effect.Sync
  import io.finch.Endpoint
  import cats.syntax.all._

  def f[F[_]: Sync]: Endpoint[F, String] = {

    val a: Endpoint[F, String] = Endpoint.param("a")
    val b: Endpoint[F, String] = Endpoint.param("a")

    a <+> b // value <+> is not a member of io.finch.Endpoint[F,String]

  }
```

This seems to be because Endpoint provides two implicit instances of `SemigroupK`, `endpointAlternative` and `endpointMonoidK`

```
ambiguous implicit values:
 both method endpointAlternative in object Endpoint of type [F[_]](implicit F: cats.MonadError[F,Throwable])cats.Alternative[[β$6$]io.finch.Endpoint[F,β$6$]]
 and method endpointMonoidK in object Endpoint of type [F[_]]=> cats.MonoidK[[β$9$]io.finch.Endpoint[F,β$9$]]
 match expected type cats.SemigroupK[X]
    SemigroupK[X].combineK(a, b)
```

### Workaround

```scala
Endpoint.endpointMonoidK.combineK(a, b)
```

**Comment by joroKr21** [comment]
Thanks for the report, that's an oversight. We should prioritize those instances by putting one of them in a parent trait or abstract class. Shouldn't be too easy to fix but I don't have access to a machine where I can fix that right now.

**Comment by custommonkey** [comment]
Thanks, I've had a go at moving the instance to a trait. #1565

**Comment by joroKr21** [comment]
Fixed by #1565
